aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/SRPCognitoActions.kt

```diff
@@ -81,6 +82,50 @@ object SRPCognitoActions : SRPActions {
             dispatcher.send(evt)
         }
 
+    override fun initiateSRPWithCustomAuthAction(event: SRPEvent.EventType.InitiateSRPWithCustom): Action =
+        Action<AuthEnvironment>("InitSRPCustomAuth") { id, dispatcher ->
+            logger?.verbose("$id Starting execution")
+            val evt = try {
+                srpHelper = SRPHelper()
+
+                val secretHash = AuthHelper.getSecretHash(
+                    event.username,
+                    configuration.userPool?.appClient,
+                    configuration.userPool?.appClientSecret
+                )
+
+                val authParams = mutableMapOf(
+                    KEY_USERNAME to event.username,
+                    KEY_SRP_A to srpHelper.getPublicA(),
+                    KEY_CHALLENGE_NAME to VALUE_SRP_A
+                )
+                secretHash?.let { authParams[KEY_SECRET_HASH] = it }
+                val encodedContextData = userContextDataProvider?.getEncodedContextData(event.username)
+
+                val initiateAuthResponse = cognitoAuthService.cognitoIdentityProviderClient?.initiateAuth {
+                    authFlow = AuthFlowType.CustomAuth
+                    clientId = configuration.userPool?.appClient
+                    authParameters = authParams
+                    encodedContextData?.let { userContextData { encodedData = it } }
+                }
+
+                when (initiateAuthResponse?.challengeName) {
+                    ChallengeNameType.PasswordVerifier -> initiateAuthResponse.challengeParameters?.let {
```

Comment:
Does this mean that second step in SRP with custom auth is always password verifier except that the sign in flow is controlled by a lambda?

Reply:
That is correct. For initiateAuth with SRP the next step should always be password verifier where it goes through the SRP flow.